packages/@aws-cdk/aws-ecs/test/fargate/fargate-task-definition.test.ts

```diff
@@ -241,8 +242,106 @@ describe('fargate task definition', () => {
+          cpuArchitecture: CpuArchitecture.X86_64,
+        },
+      });
 
+      // THEN
+      expect(stack).toHaveResourceLike('AWS::ECS::TaskDefinition', {
+        Cpu: '1024',
+        Family: 'FargateTaskDef',
+        Memory: '2048',
+        NetworkMode: 'awsvpc',
+        RequiresCompatibilities: [
+          ecs.LaunchType.FARGATE,
+        ],
+        RuntimePlatform: {
+          CpuArchitecture: 'X86_64',
+          OperatingSystemFamily: 'WINDOWS_SERVER_2019_CORE',
+        },
+        TaskRoleArn: {
+          'Fn::GetAtt': [
+            'FargateTaskDefTaskRole0B257552',
+            'Arn',
+          ],
+        },
+      });
     });
+
+    test('runtime testing for windows container', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+      new ecs.FargateTaskDefinition(stack, 'FargateTaskDef', {
+        cpu: 1024,
+        memoryLimitMiB: 2048,
+        runtimePlatform: {
+          operatingSystemFamily: OperatingSystemFamily.LINUX,
+          cpuArchitecture: CpuArchitecture.ARM64,
+        },
+      });
+
+      // THEN
+      expect(stack).toHaveResourceLike('AWS::ECS::TaskDefinition', {
+        Cpu: '1024',
+        Family: 'FargateTaskDef',
+        Memory: '2048',
+        NetworkMode: 'awsvpc',
+        RequiresCompatibilities: [
+          ecs.LaunchType.FARGATE,
+        ],
+        RuntimePlatform: {
+          CpuArchitecture: 'ARM64',
+          OperatingSystemFamily: 'LINUX',
+        },
+        TaskRoleArn: {
+          'Fn::GetAtt': [
+            'FargateTaskDefTaskRole0B257552',
+            'Arn',
+          ],
+        },
+      });
+    });
+
+    test('creating a Fargate TaskDefinition with WINDOWS_SERVER_X operatingSystemFamily and incorrect cpu throws an error', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // Not in CPU Ranage.
+      expect(() => {
+        new ecs.FargateTaskDefinition(stack, 'FargateTaskDefCPU', {
+          cpu: 128,
+          memoryLimitMiB: 1024,
+          runtimePlatform: {
+            cpuArchitecture: ecs.CpuArchitecture.X86_64,
+            operatingSystemFamily: ecs.OperatingSystemFamily.WINDOWS_SERVER_2019_CORE,
+          },
+        });
+      }).toThrowError(`If operatingSystemFamily is ${ecs.OperatingSystemFamily.WINDOWS_SERVER_2019_CORE._operatingSystemFamily}, then cpu must be in 1024 (1 vCPU), 2048 (2 vCPU), or 4096 (4 vCPU).`);
+
+      // Memory is not in 1 GB increments.
+      expect(() => {
+        new ecs.FargateTaskDefinition(stack, 'FargateTaskDefMemory', {
+          cpu: 1024,
+          memoryLimitMiB: 1025,
+          runtimePlatform: {
+            cpuArchitecture: ecs.CpuArchitecture.X86_64,
+            operatingSystemFamily: ecs.OperatingSystemFamily.WINDOWS_SERVER_2019_CORE,
+          },
+        });
+      }).toThrowError('If define vCPU equal 1024, Memory need in Min. 2GB and Max. 8GB, in 1GB increments, cannot define 1025.');
+
+    });
+
```

Comment:
Can you include one more test here that checks that the right error is thrown if `cpu` and `memoryLimitMiB` are both undefined, but `runtimePlatform` is? I believe the error should be: `Fargate-compatible tasks require both CPU (${props.cpu}) and memory (${props.memoryMiB}) specifications`

Reply:
Hi @madeline-k , I found this source code.
https://github.com/aws/aws-cdk/blob/cac11bba2ea0714dec8e23b069496d1b9d940685/packages/%40aws-cdk/aws-ecs/lib/fargate/fargate-task-definition.ts#L124-L131

if `props.cpu` or `props.memoryMiB` is undefined, I think it will been defined to default value right?!

wdyt @pahud

Reply:
You're right, if someone creates a new FargateTaskDefinition, there will be those default values set. But if they create a new TaskDefinition, and specify a runtimePlatform, but no memory and cpu, that should cause an error. You can add a test to `task-definition.test.ts`. But I can see the argument that that is out of scope for this PR, so up to you. If you decide to opt to not include that, let me know and I can approve. Thanks for making these latest updates!

Reply:
Hi @madeline-k,thanks for review. 
I will create another PR for that testing. 

Let's let this PR approved.😆